build.rs

```diff
@@ -67,9 +67,145 @@
+                                            "crypto/rand/sysrand.c",
+                                            "crypto/rsa/blinding.c",
+                                            "crypto/rsa/rsa.c",
+                                            "crypto/rsa/rsa_impl.c"];
+
+const RING_INTEL_SHARED_SRCS: &'static [&'static str] = &["crypto/cpu-intel.c"];
+
+const RING_X86_SRCS: &'static [&'static str] =
+    &["crypto/aes/asm/aes-586.pl",
+      "crypto/aes/asm/aesni-x86.pl",
+      "crypto/aes/asm/vpaes-x86.pl",
+      "crypto/bn/asm/x86-mont.pl",
+      "crypto/chacha/asm/chacha-x86.pl",
+      "crypto/ec/asm/ecp_nistz256-x86.pl",
+      "crypto/modes/asm/ghash-x86.pl",
+      "crypto/poly1305/asm/poly1305-x86.pl",
+      "crypto/sha/asm/sha256-586.pl",
+      "crypto/sha/asm/sha512-586.pl"];
+
+const RING_X86_64_SRC: &'static [&'static str] =
+    &["crypto/aes/asm/aes-x86_64.pl",
+      "crypto/aes/asm/aesni-x86_64.pl",
+      "crypto/aes/asm/bsaes-x86_64.pl",
+      "crypto/aes/asm/vpaes-x86_64.pl",
+      "crypto/bn/asm/rsaz-avx2.pl",
+      "crypto/bn/asm/x86_64-mont.pl",
+      "crypto/bn/asm/x86_64-mont5.pl",
+      "crypto/chacha/asm/chacha-x86_64.pl",
+      "crypto/curve25519/asm/x25519-asm-x86_64.S",
+      "crypto/curve25519/x25519-x86_64.c",
+      "crypto/ec/asm/ecp_nistz256-x86_64.pl",
+      "crypto/ec/asm/p256-x86_64-asm.pl",
+      "crypto/modes/asm/aesni-gcm-x86_64.pl",
+      "crypto/modes/asm/ghash-x86_64.pl",
+      "crypto/poly1305/asm/poly1305-x86_64.pl",
+      "crypto/sha/asm/sha256-x86_64.pl",
+      "crypto/sha/asm/sha512-x86_64.pl"];
+
+const RING_ARM_SHARED_SRCS: &'static [&'static str] =
+    &["crypto/cpu-arm.c",
+      "crypto/cpu-arm-linux.c",
+      "crypto/aes/asm/aesv8-armx.pl",
+      "crypto/modes/asm/ghashv8-armx.pl"];
+
+const RING_ARM_SRCS: &'static [&'static str] =
+    &["crypto/aes/asm/aes-armv4.pl",
+      "crypto/aes/asm/bsaes-armv7.pl",
+      "crypto/bn/asm/armv4-mont.pl",
+      "crypto/chacha/asm/chacha-armv4.pl",
+      "crypto/curve25519/asm/x25519-asm-arm.S",
+      "crypto/ec/asm/ecp_nistz256-armv4.pl",
+      "crypto/modes/asm/ghash-armv4.pl",
+      "crypto/poly1305/asm/poly1305-armv4.pl",
+      "crypto/sha/asm/sha256-armv4.pl",
+      "crypto/sha/asm/sha512-armv4.pl"];
+
+const RING_AARCH64_SRCS: &'static [&'static str] =
+    &["crypto/cpu-aarch64-linux.c",
+      "crypto/bn/asm/armv8-mont.pl",
+      "crypto/chacha/asm/chacha-armv8.pl",
+      "crypto/ec/asm/ecp_nistz256-armv8.pl",
+      "crypto/poly1305/asm/poly1305-armv8.pl",
+      "crypto/sha/asm/sha256-armv8.pl",
+      "crypto/sha/asm/sha512-armv8.pl"];
+
+const RING_TEST_SRCS: &'static [&'static str] =
+    &["crypto/bn/bn_test.cc",
+      "crypto/constant_time_test.c",
+      "crypto/test/bn_test_convert.c",
+      "crypto/test/bn_test_lib.c",
+      "crypto/test/file_test.cc"];
+
+const RING_HEADERS: &'static [&'static str] = &["crypto/poly1305/internal.h",
```

Comment:
This would be impossible to keep in sync with the source if we have to do it 100% manually. Already, the `*.inl` files are missing from this list, and so is `find crypto/perlasm -name "*.pl"` is missing, at least.

On the other hand, we can't have build.rs build the list dynamically from doing something like `find crypto include mk -name "*.h" -o -name "*.inl" -o -name "*.mk"` itself, because that won't account for removed files.

Maybe we should just always run the build script, and have the build script do conditional rebuild logic on a per-source-file so that it isn't ever too slow? After all, we want an incremental build where only a `*.c` file changed to be about as fast as a build where only a `*.rs` file changed. If the difference between running build.rs or not is large enough to be worth skipping then build.rs isn't fast enough.

WDYT?